source/PlayerInfo.cpp

```diff
@@ -4165,6 +4166,15 @@ void PlayerInfo::StepMissions(UI *ui)
 
 
 
+void PlayerInfo::StepMissionTimers(UI *ui)
+{
+	for(Mission &mission : missions)
+		for(Timer &timer : mission.Timers())
+			timer.Step(*this, ui, mission);
```

Comment:
if we would move this into mission, player would not have to touch the timer class

Reply:
So just to explain this, I think if possible double sided includes should be avoided, in this case player uses timer and timer uses player, suboptimal imo

Reply:
I fully approve of the principle; however, I'm still having a little trouble implementing this—specifically, in `MainPanel.cpp:106` where `PlayerInfo::StepMissionTimers()` is called, I would need to replace it with iterating over the Missions in the `player` object, then calling a new `StepTimers()` method on each one. So far so good, but `PlayerInfo::Missions()` returns a list of `const Mission &` objects, and `StepTimers()` is a mutating method, so if I understand correctly, it cannot be `const`, and thus cannot be called on a `const Mission &`.

Reply:
That's not what I meant, keep this function but loop through the timers inside the mission class